extensions/amp-story/1.0/utils.js

```diff
@@ -323,3 +323,22 @@ export function triggerClickFromLightDom(anchorElement, domElement) {
   outerAnchor.click();
   outerAnchor.remove();
 }
+
+/**
+ * Makes a proxy URL if document is served from a proxy origin. No-op otherwise.
+ * @param {string} url
+ * @param {!Element} pageEl
+ * @return {string}
+ */
+export const maybeMakeProxyUrl = (url, pageEl) => {
+  const urlService = Services.urlForDoc(pageEl);
+  const loc = pageEl.getAmpDoc().win.location;
```

Comment:
Now that it's moved to `utils.js` I think we should make it easier to re-use, and accept `url, ampdoc` as parameters.
`urlForDoc(ampdoc)` should work, and both classes using this helper should be able to call `this.getAmpDoc()`.